### DISCUSSION TITLE
Issue with using inside docker container

### DISCUSSION THREAD

**Discussion by BaluVyamajala** [LukeCarrier/mkdocs-drawio-exporter#68] [Q&A] [answered]
Trying to setup a development server for running mkdocs locally. so, users don't have to install draw io in their mac/windows systems. But several things I have tried and finally left at below configuration which goes into infinate loop trying to export images for draw.io?

Here is my Docker file:

```
from python:3.8

WORKDIR /app


ARG drawio_arch=amd64
ARG drawio_version=18.1.3
ARG drawio_deb="drawio-${drawio_arch}-${drawio_version}.deb"
ARG drawio_url="https://github.com/jgraph/drawio-desktop/releases/download/v${drawio_version}/${drawio_deb}"

RUN curl -L -o "$drawio_deb" "$drawio_url"
RUN apt-get update && \
    apt-get install -y libasound2 xvfb ./"$drawio_deb"

EXPOSE 8000

COPY requirements.txt .
RUN pip install -r requirements.txt 
COPY . /docs/
COPY ./mkdocs.yml . 

ENTRYPOINT ["xvfb-run", "mkdocs"]
CMD ["serve"]
```

docker-compose.yaml

```
services:
  web:
    build: .
    volumes:
      - ./docs:/app/docs
      - ./mkdocs.yml:/app/mkdocs.yml
    ports:
      - "8000:8000"
    privileged: true
    init: true
```

extensions:
```
markdown_extensions:
  - tables
  - attr_list
plugins:
  - drawio-exporter:
      cache_dir: 'assets/drawings/drawio-exporter'
      drawio_args:
        - --no-sandbox
```

<img width="1321" alt="image" src="https://github.com/LukeCarrier/mkdocs-drawio-exporter/assets/23091268/01ad6275-cb61-4b47-ab0a-7661bd328dd0">

Any help would be greatly appreciated.

**Comment by LukeCarrier** [comment]
Did you read [the readme](https://github.com/LukeCarrier/mkdocs-drawio-exporter?tab=readme-ov-file#running-without-the-sandbox)?

**Reply by BaluVyamajala** [reply]
yes i read that, i also looked into that github action example provided and based on those i built this image for development, 
I added

- ` - --no-sandbox` option in drawio_args 
- Using `xvfb-run mkdocs serve`to start the container. (entrypoint + cmd)

**Reply by LukeCarrier** [reply]
`xvfb-run` is unrelated -- it sets up the X virtual framebuffer required for running graphical applications.

If you pass `mkdocs` the `-v` (or `--verbose`) option do you see `--no-sandbox` being passed to `drawio`? The output should look something like (edited out a lot of noise):

```console
❯ poetry run mkdocs -v serve

[snip]

DEBUG   -  Running `config` event from plugin 'drawio-exporter'
DEBUG   -  Trying paths ['/Users/lukecarrier/Applications/draw.io.app/Contents/MacOS/draw.io',
           '/Applications/draw.io.app/Contents/MacOS/draw.io'] for platform "darwin"
DEBUG   -  Found Draw.io executable for platform "darwin" at
           "/Applications/draw.io.app/Contents/MacOS/draw.io"
DEBUG   -  Using Draw.io executable "/Applications/draw.io.app/Contents/MacOS/draw.io", arguments
           ['--no-sandbox'] and cache directory
           "[snip]/docs/drawio-exporter"

[snip]

DEBUG   -  Running `post_page` event from plugin 'drawio-exporter'
DEBUG   -  Exporting
           "[snip]/docs/infrastructure/deployment.drawio"
           to
           "[snip]/docs/drawio-exporter/4b031b9a3885e4aedef021845d0b83f3eace2de6-0"
DEBUG   -  Using export command ['/Applications/draw.io.app/Contents/MacOS/draw.io', '--export',
           '[snip]/docs/infrastructure/deployment.drawio',
           '--page-index', '0', '--output',
           '[snip]/docs/drawio-exporter/4b031b9a3885e4aedef021845d0b83f3eace2de6-0',
           '--format', 'svg', '--no-sandbox']
Exporting  [
  '[snip]/docs/infrastructure/deployment.drawio'
]
[snip]/docs/infrastructure/deployment.drawio -> [snip]/docs/drawio-exporter/4b031b9a3885e4aedef021845d0b83f3eace2de6-0

[snip]
```

Configuration:

```yaml
plugins:
  - search
  # snip
  - drawio-exporter:
      embed_format: '{content}'
      drawio_args: ['--no-sandbox']
```

If you don't, it's likely a configuration issue. If you do, it's possible Draw.io Desktop isn't passing the option to Chromium, and it can't be fixed here.

In any case, I'd strongly advise you to avoid running the process as root. I will try and test this with recent Draw.io versions and throw an example together at some point.

**Reply by LukeCarrier** [reply]
See the example in #69 as a base.

**Comment by LukeCarrier** [comment, marked as answer]
Put together an example that seems to work with recent Draw.io Desktop builds in #69.
